components/brave_referrals/browser/brave_referrals_service.cc

```diff
@@ -115,13 +131,76 @@ std::string BuildReferralEndpoint(const std::string& path) {
   if (env->HasVar("BRAVE_REFERRALS_LOCAL"))
     proto = "http";
 
-  return base::StringPrintf("%s://%s%s", proto.c_str(),
-                            referral_server.c_str(),
+  return base::StringPrintf("%s://%s%s", proto.c_str(), referral_server.c_str(),
                             path.c_str());
 }
 
 }  // namespace
 
+BraveReferralsHeaders::BraveReferralsHeaders() {
+  // NOTE TO READER: This header is for partners to determine the browser is
+  // Brave without creating webcompat issues by creating a new user-agent.
+  // All Brave users send the exact same header and so this cannot be used for
+  // tracking individual users. See
+  // https://github.com/brave/brave-browser/wiki/Custom-Headers for more
+  // information. Custom headers are deprecated and new partners use the
+  // navigator.brave.isBrave() JavaScript API.
+  constexpr char kPartnerEaffName[] = "eaff";
+  constexpr char kPartnerUpholdName[] = "uphold";
+  constexpr char kPartnerGrammarlyName[] = "grammarly";
+
+  referral_headers_.push_back(
```

Comment:
personally, I would make global
```
constexpr auto kHeaders = base::MakeFixedFlatMap<base::StringPiece, base::StringPiece> {
 {"eaff.com", "eaff}, {"stg.eaff.com", "eaff"} , {"uphold.com", "uphold}, ...
}
```
then I would iterate over keys and match the domain and then get the header value in one line (if this is everything we need).